nixos/modules/services/audio/librespot.nix

```diff
@@ -0,0 +1,205 @@
+        type = types.bool;
+        default = false;
+        description = "Enables linear volume scaling instead of logarithmic (default) scaling";
+      };
+      disable-gapless = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Disables gapless playback by forcing the sink to close b/w tracks";
+      };
+      zeroconf-port = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "The port that the HTTP server advertised on zeroconf will use [1-65535]. Ports <= 1024 may require root priviledges.";
+      };
+      proxy = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Use a proxy for resolving the access point. Proxy should be an HTTP proxy in the form http://ip:port, and can also be passed using the all-lowercase http_proxy environment variable.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [ librespot ];
+
+    systemd.services.librespot = {
+      description = "librespot";
+      after    = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+      ExecStart = "${pkgs.librespot}/bin/librespot -n ${cfg.name} "
+        + optionalString (cfg.cache != null) "--cache ${cfg.chache} "
+        + optionalString (cfg.disable-audio-cache) "--disable-audio-cache "
+        + optionalString (cfg.device-type != null) "--device-type ${cfg.device-type} "
+        + optionalString (cfg.bitrate != null) "-b ${toString cfg.bitrate} "
+        + optionalString (cfg.onevent != null) "--onevent ${cfg.onevent} "
+        + optionalString (cfg.verbose) "--verbose "
+        + optionalString (cfg.username != null) "--username ${cfg.username} "
+        + optionalString (cfg.password != null) "--password ${cfg.password} "
+        + optionalString (cfg.disable-discovery) "--disable-discovery "
+        + optionalString (cfg.device != null) "--device ${cfg.device} "
+        + optionalString (cfg.mixer != null) "--mixer ${cfg.mixer} "
+        + optionalString (cfg.mixer-name != null) "--mixer-name ${cfg.mixer-name} "
+        + optionalString (cfg.mixer-card != null) "--mixer-card ${cfg.mixer-card} "
+        + optionalString (cfg.mixer-index != null) "--mixer-index ${toString cfg.mixer-index} "
+        + optionalString (cfg.inital-volume != null) "--inital-volume ${toString cfg.inital-volume} "
+        + optionalString (cfg.enable-volume-normalisation) "--enable-volume-normalisation "
+        + optionalString (cfg.normalisation-pregain != null) "--normalisation-pregain ${toString cfg.normalisation-pregain} "
+        + optionalString (cfg.linear-volume) "--linear-volume "
+        + optionalString (cfg.disable-gapless) "--disable-gapless "
+        + optionalString (cfg.zeroconf-port != null) "--zeroconf-port ${toString cfg.zeroconf-port} "
+        + optionalString (cfg.proxy != null) "--proxy ${toString cfg.proxy} "
```

Comment:
Most of these arguments should be escaped via `escapeShellArg`.
Also `backend` option would be great.
Consider this example: `librespot -n "Librespot Speaker" -b 160 --backend subprocess --device "ffplay -i -  -f s16le -ac 2 -nodisp"`